### PR TITLE
Changes for CAA records feature in Azure DNS

### PR DESCRIPTION
**Pull request by karravi-msft**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ x] Title of the pull request is clear and informative.
- [ x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x ] Pull request includes test coverage for the included changes.
- [ x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Dns/Commands.Dns/Records/AddAzureDnsRecordConfig.cs [hunk 1]
```diff
@@ -80,6 +82,21 @@ public class AddAzureDnsRecordConfig : DnsBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Cname { get; set; }
 
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The flags for the CAA record to add. Must be a number between 0 and 255.", ParameterSetName = ParameterSetCaa)]
+        [ValidateNotNullOrEmpty]
+        public byte CaaFlags { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The tag field of the CAA record to add.", ParameterSetName = ParameterSetCaa)]
+        [ValidateNotNullOrEmpty]
+        public string CaaTag { get; set; }
```
Comment (nkindberg) [inline src/ResourceManager/Dns/Commands.Dns/Records/AddAzureDnsRecordConfig.cs]: Tag should be 1-15 characters if you want to add length validation here

File: src/ResourceManager/Dns/Commands.Dns/Records/RemoveAzureDnsRecordConfig.cs [hunk 2]
```diff
@@ -78,6 +80,20 @@ public class RemoveAzureDnsRecordConfig : DnsBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Cname { get; set; }
 
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The flags for the CAA record to add. Must be a number between 0 and 255.", ParameterSetName = ParameterSetCaa)]
+        [ValidateNotNullOrEmpty]
+        public byte CaaFlags { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The tag field of the CAA record to add.", ParameterSetName = ParameterSetCaa)]
+        [ValidateNotNullOrEmpty]
+        public string CaaTag { get; set; }
```
Comment (nkindberg) [inline src/ResourceManager/Dns/Commands.Dns/Records/RemoveAzureDnsRecordConfig.cs]: 1-15 length for tag

File: src/ResourceManager/Dns/Commands.Dns/Records/NewAzureRmDnsRecordConfig.cs [hunk 3]
```diff
@@ -88,6 +89,20 @@ public class NewAzureRmDnsRecordConfig : DnsBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Cname { get; set; }
 
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The flags for the CAA record to add. Must be a number between 0 and 255.", ParameterSetName = ParameterSetCaa)]
+        [ValidateNotNullOrEmpty]
+        public byte CaaFlags { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The tag field of the CAA record to add.", ParameterSetName = ParameterSetCaa)]
+        [ValidateNotNullOrEmpty]
+        public string CaaTag { get; set; }
```
Comment (nkindberg) [inline src/ResourceManager/Dns/Commands.Dns/Records/NewAzureRmDnsRecordConfig.cs]: Same comment here - Tag should be 1 to 15 if you want to add length validator

File: src/ResourceManager/Dns/Commands.Dns/Records/RemoveAzureDnsRecordConfig.cs (hunk 2 above)
Comment (karravi-msft) [reply src/ResourceManager/Dns/Commands.Dns/Records/RemoveAzureDnsRecordConfig.cs]: I will skip this validation for now...

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/121/

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/124/

**Pull request merged**